dev/src/reference.ts

```diff
@@ -2285,7 +2285,9 @@ export class Query<T = firestore.DocumentData> implements firestore.Query<T> {
           if (proto.done) {
             logger('Query._stream', tag, 'Trigger Logical Termination.');
             backendStream.unpipe(stream);
-            stream.emit('end');
+            backendStream.resume();
```

Comment:
Why do we need to `resume` and then `end` ? by reading the [documentation](https://sourcegraph.com/npm/types/node@a90b5bfc0c1a1602b1dc1bf6fffdb44a889c1a46/-/blob/stream.d.ts?L305&subtree=true) it seems that `unpipe` followed by `end` should be a reasonable thing to do?

Reply:
There is some funky logic that requires the 'end' event to emit.

https://github.com/googleapis/nodejs-firestore/blob/main/dev/src/index.ts#L1623-L1623

For some reason, this doesn't happen unless the backendStream is resumed.

I don't see where in your referenced documentation it shows 'end' doesn't require 'resume'. As far as I can tell, documentation only shows calling 'end' on the downstream, not upstream.

Reply:
👍